test/stdlib/json/JSON_test.rb

```diff
@@ -238,4 +238,58 @@ def test_unparse
     assert_send_type "(ToJson) -> String", MyJSON.new, :unparse, ToJson.new
     assert_send_type "(ToJson, indent: String) -> String", MyJSON.new, :unparse, ToJson.new, { indent: "\t" }
   end
+
+  def test_to_json_with_object
+    assert_send_type "() -> String", Object.new, :to_json
+    assert_send_type "(JSON::State) -> String",  Object.new, :to_json, JSON::State.new
+    assert_send_type "(nil) -> String", Object.new, :to_json, nil
+  end
```

Comment:
[IMO] I prefer the multiline style to reduce cognitive load, but I follow the existing style in this file.

```diff
-assert_send_type "() -> String", Object.new, :to_json
+assert_send_type "() -> String",
+                 Object.new, :to_json
```

Reply:
I prefer the multiline style too, but don't have strong opinion to fix all of the existing test code.